ui/src/app/pages/homepage/homepage.tsx

```diff
@@ -156,6 +164,25 @@ export const Homepage = withUserProfile()(class extends React.Component<Props, S
     }
   }
 
+  async validateRasLoginGovLink() {
+    const authCode = (new URL(window.location.href)).searchParams.get('code');
+    const redirectUrl = buildRasRedirectUrl();
+    if (authCode) {
+      this.setState({rasLoginGovLoading: true});
+      try {
+        const profileResponse = await profileApi().linkRasAccount({ authCode, redirectUrl });
+        if (profileResponse.rasLinkLoginGovUsername !== undefined) {
+          this.setState({rasLoginGovLinked: true});
+        }
+      } catch (e) {
+        this.setState({rasLoginGovLinkError: 'Error saving RAS Login.Gov linkage status.'});
+        this.setState({rasLoginGovLoading: false});
+      }
+    }
+    // Cleanup parameter from URL after linking.
+    window.history.pushState({}, '', '/');
```

Comment:
`replaceState` may be be better here. I am not sure what the url looks like before replacing, but if the router cannot resolve it and the user clicks the back button the router may throw an error.

Also, should this be inside of the conditional block?

Reply:
Thanks! Changed to replaceState.
(2) This should be outside of conditaion block because we want to clean up the url no matte the link succeed or not